### ISSUE TITLE
Current dev frontend high CPU load and unusable -- probably history related

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#10001]
This is on ae77bcc3268595737a9c090e78bd169b7851b95d.
Nothing can be clicked and the CPU is at 100% until I close the tab, and the collection status message is weirdly shifted.

<img width="159" alt="Screenshot 2020-07-15 at 13 04 17" src="https://user-images.githubusercontent.com/6804901/87537926-be1ca400-c69b-11ea-9ca4-49da350eb3f8.png">

While I can still scroll I see this:
<img width="311" alt="Screenshot 2020-07-15 at 13 09 35" src="https://user-images.githubusercontent.com/6804901/87538531-a72a8180-c69c-11ea-99d4-290313b9442f.png">

On the backend I also see 100%CPU load for a minute or so, this is the stack:
```
    serialize (galaxy/managers/base.py:610)
    serialize (galaxy/managers/datasets.py:612)
    serialize (galaxy/managers/hdas.py:372)
    serialize_to_view (galaxy/managers/base.py:692)
    __index_v2 (galaxy/webapps/galaxy/api/history_contents.py:922)
    index (galaxy/webapps/galaxy/api/history_contents.py:82)
    decorator (galaxy/web/framework/decorators.py:295)
    handle_request (galaxy/web/framework/base.py:224)
    __call__ (galaxy/web/framework/base.py:145)
    __call__ (httpexceptions.py:640)
    __call__ (recursive.py:85)
    __call__ (galaxy/web/framework/middleware/error.py:154)
    __call__ (galaxy/web/framework/middleware/translogger.py:71)
    __call__ (galaxy/web/framework/middleware/xforwardedhost.py:23)
    __call__ (galaxy/web/framework/middleware/request_id.py:15)
    __call__ (galaxy/web/framework/middleware/batch.py:80)
```
It seems we're serializing datasets here, despite very few datasets being in the top level history.

**Comment by mvdbeek** [comment]
<img width="1339" alt="Screenshot 2020-07-15 at 13 22 39" src="https://user-images.githubusercontent.com/6804901/87539549-603d8b80-c69e-11ea-8a50-aed06aeeb1b1.png">
I suppose we're returning too many items in that query and that is too much for the browser to keep up with ?
@Nerdinacan any idea what might be up with this ?

**Comment by mvdbeek** [comment]
If I add a visible=True to that one giant query I get an immediate response from the API, and I think that should fix it, if I can find out where we do the query.

**Comment by mvdbeek** [comment]
The weird shift and scrolling is still there when there is a lot of new datasets:
![weird](https://user-images.githubusercontent.com/6804901/94584464-f1c0a080-027e-11eb-8c62-d726104fdfb3.gif)
But I don't see the giant response anymore, so I guess it's not critical to get this fixed for 20.09

**Comment by mvdbeek** [comment]
Haven't seen this anymore
